lib/files/node.ts

```diff
@@ -153,6 +153,10 @@ export abstract class Node {
 	 * Get the absolute path of this object relative to the root
 	 */
 	get path(): string|null {
+		// if the current node IS the root, return /
+		if (this.root && this.source.endsWith(this.root)) {
```

Comment:
Hum, I don't like this check, if someone create a folder that have the same path as the root, that would still match

```js
source: 'https://domain.com/remote.php/dav/files/emma/files/emma/'
root: '/files/emma'
```

Reply:
![image](https://user-images.githubusercontent.com/14975046/216581378-f7cd1fb6-1db5-493b-bc15-6eecde331e4d.png)

Reply:
Fixed, I added some sanity checks, and some more testing use cases.